### PR TITLE
Class-based implementation of query routes

### PR DESCRIPTION
**Pull request by brimoor**
The rest of the codebase uses class-like style, so I'd like to stick with that syntax for now.

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -157,6 +189,22 @@ class API {
     return JSON.parse(body).jobs;
   }
 
+  /**
+   * Performs a customized jobs query.
+   *
+   * @async
+   * @param {JobsQuery} jobsQuery An JobsQuery instance defining the customzied
```
Comment (hoddr) [inline lib/api.js]: Should be `A JobsQuery` instead of `An JobsQuery`

Comment (hoddr) [reply lib/api.js]: Also `customzied -> customized`

File: lib/query.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    limit = Number.parseInt(limit);
+    if (!isNaN(limit) && limit > 0) {
+      this.limit = limit;
+    }
+    return this;
+  }
+
+  /**
+   * Converts the query instance into an object suitable for passing to the
+   * `request` library.
```
Comment (hoddr) [inline lib/query.js:116]: Are we passing this to `request` or to `qs` (being a bit pedantic)?

**Review by hoddr — APPROVED** [review]
LGTM after addressing minor comments.

File: lib/query.js (hunk 2 above)
Comment (brimoor) [reply lib/query.js:116]: we use `qs` to support `toString`, but when we actually send the query, we send directly with `request`

**Pull request merged**